napari/__main__.py

```diff
@@ -6,13 +6,31 @@
 
 import numpy as np
 
-from .utils import io
-from . import Viewer, gui_qt
+from . import Viewer, __version__, gui_qt
+from .utils import io, sys_info
+
+
+class InfoAction(argparse.Action):
+    def __call__(self, *args, **kwargs):
+        print(sys_info())
+        sys.exit()
```

Comment:
Is there an advantage to making this a class instead of a function?

Reply:
don't really "want" it to be a class, but looked into it briefly and actually not sure what the best way to run a function and quit would be without using the [`argparse.Action` protocol](https://docs.python.org/3/library/argparse.html#argparse.Action), which only described doing it with a subclass.  Do you have an alternate strategy?

Reply:
@tlambert03 I think this works?

```python
def info_action(*args, **kwargs):
    print(sys_info())
    sys.exit()

...

    action=info_action)
```

Reply:
actually... this appears to work at first, but I think `info_action` _always_ gets called.  presumably, the `action` argument of `parser.add_argument` is expecting a class, not a function, and will call it to instantiate it (only later calling `instance.__call__` if the `--flag` is `True`) so giving it a function just calls the function immediately and exits the program regardless of other arguments...  I tried some other function strategies, but couldn't get it to work.   Will leave it as a class unless you have other ideas.